src/test/java/dev/streamx/cli/ingestion/publish/PublishPayloadCommandTest.java

```diff
@@ -0,0 +1,272 @@
+package dev.streamx.cli.ingestion.publish;
+
+import static com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder.responseDefinition;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.matchingJsonPath;
+import static com.github.tomakehurst.wiremock.client.WireMock.put;
+import static com.github.tomakehurst.wiremock.client.WireMock.putRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.common.ContentTypes.APPLICATION_JSON;
+import static com.github.tomakehurst.wiremock.common.ContentTypes.CONTENT_TYPE;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static dev.streamx.clients.ingestion.StreamxClient.PUBLICATIONS_ENDPOINT_PATH_V1;
+import static org.apache.hc.core5.http.HttpStatus.SC_ACCEPTED;
+import static org.apache.hc.core5.http.HttpStatus.SC_OK;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.common.Json;
+import com.github.tomakehurst.wiremock.junit5.WireMockExtension;
+import com.github.tomakehurst.wiremock.matching.ContainsPattern;
+import dev.streamx.clients.ingestion.publisher.PublisherSuccessResult;
+import io.quarkus.test.junit.main.LaunchResult;
+import io.quarkus.test.junit.main.QuarkusMainLauncher;
+import io.quarkus.test.junit.main.QuarkusMainTest;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@QuarkusMainTest
+public class PublishPayloadCommandTest {
+  private static final String CHANNEL = "pages";
+  private static final String KEY = "index.html";
+  private static final String DATA = """
+        {"content": {"bytes": "<h1>Hello World!</h1>"}}""";
+
+  @RegisterExtension
+  static WireMockExtension wm = WireMockExtension.newInstance()
+      .options(wireMockConfig().dynamicPort()).configureStaticDsl(true).build();
+
+  @BeforeEach
+  void setup() {
+    initializeWiremock();
+  }
+
+  @Test
+  public void shouldRejectInvalidJsonPath(QuarkusMainLauncher launcher) {
```

Comment:
I believe that we should add all non existing values/replacements to payload instead of throwing an expection.